### ISSUE TITLE
Semantic version required

### ISSUE THREAD

**Issue by tiancaiamao** [pingcap/parser#69]
We've make the parser a standalone repository, moving the code out from [TiDB](https://github.com/pingcap/tidb). One of the main purpose is to make the parser more general one.

The parser should be general enough to work as a library, works in both TiDB 2.1 and TiDB and many other projects (maybe different module version).

To reach that goal, I think it's time to tag release version for the parser, and employ the [sematic version](https://semver.org/).

**Comment by bb7133** [comment]
Hi @tiancaiamao , shall we make your issue move forward?

**Comment by tiancaiamao** [comment]
> Hi @tiancaiamao , shall we make your issue move forward?

No, we can't, as long as we continue using the current branch and version management policy.
Semantic version forbids incompatible changes for Minor version, while we're not using the concept of version, we just use branches.
